### ISSUE TITLE
In training classification task which is step 2 for the process. 

### ISSUE THREAD

**Issue by Shield9999** [HAL-42/FMA-WSSS#33]
In training classification task which is step 2 for the process I get weight missed for the pascal and coco. Any solution for this? Or is this the right one.

 
![image](https://github.com/HAL-42/FMA-WSSS/assets/110029253/a6c87229-894b-4331-aa98-17c50590456b)

**Comment by HAL-42** [comment]
This shouldn't be a problem. The missing parameter is CLIP's pre-trained weights.

**Comment by Shield9999** [comment]
Thx for your comment!

**Comment by Shield9999** [comment]
To Yang Xiaobo

Hello author. I'm using your foundation model as my baseline to develop the
performance. When I'm reproducing your baseline performance I could
reproduce the COCO performance however for the pascal voc there are some
gap between the experiment and your paper. 79.5 or 81.2 for the validation
set, and the test set was fine. Could tell me more detail or why this
performance gap is happening?

Best regards,
David Jung(SHIELD 999)

2024년 7월 1일 (월) 오후 3:30, Yang Xiaobo ***@***.***>님이 작성:

**Comment by HAL-42** [comment]
[Here](https://zjueducn-my.sharepoint.com/personal/hal_42_zju_edu_cn/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fhal%5F42%5Fzju%5Fedu%5Fcn%2FDocuments%2FTrans%2DWSSS%2DOS&ga=1) , all experimental results and checkpoints are provided. By comparing logs and results step by step, can you locate at which specific step the difference occurred?

**Comment by Shield9999** [comment]
To. Yang Xiaobo

The experimental result on the step 'Train segmentation network with fine
seed' part which command is
bash tools/dist_train.sh configs/m2f_psudo/m2f-sl22-bt4-80k-512x-VOC.py 2'.
At this step I used my seed and use author's provided seed too. However for
me the best result was 81.3 for the validation step and for the author's
seed the best mIoU was 81.73. However the best mIoU in the paper was 83.1.
I want to reproduce this result. Could you give me some advice?

Best regards
David Jung


2024년 7월 18일 (목) 오후 10:11, Yang Xiaobo ***@***.***>님이 작성:

**Comment by HAL-42** [comment]
The best mIoU appeared at step=30000, not the final step. MMSeg should save a checkpoint named `best_mIoU_iter_30000.pth`.
Diff your log with provied `experiment\others\mmseg\m2f-sl22-bt4-80k-512x-VOC\20231208_223350\20231208_223350.log`
